crates/relayer-cosmos/src/base/impls/chain.rs

```diff
@@ -119,18 +119,7 @@ where
         &self,
         messages: Vec<CosmosIbcMessage>,
     ) -> Result<Vec<Vec<Event>>, Error> {
-        let signer = self.chain.signer();
-
-        let raw_messages = messages
-            .into_iter()
-            .map(|message| (message.to_protobuf_fn)(signer).map_err(Error::encode))
-            .collect::<Result<Vec<_>, _>>()?;
-
-        let events = simple_send_tx(self.chain.tx_config(), self.chain.key_entry(), raw_messages)
-            .await
-            .map_err(Error::relayer)?;
-
-        Ok(events)
+        self.tx_context.send_messages(messages).await
```

Comment:
The first iteration for sending chain messages using the new transaction context is completed. With this, we no longer need to rely on the legacy functions like `simple_send_tx` to send transactions. Moving forward, we will be able to customize how the transactions should be sent inside the relayer framework itself.